### PR TITLE
Update example in "Rust Inside Other Languages"

### LINKED ISSUE #25855: Code in book's example thread not executed
I'm running the example from the ["Rust Inside Other Languages"][1] section in Rust Book.

I noticed that when I embed the code into Python and execute it, the execution time does not depend on the counter size (initially set to 5 million in the book). Although the text suggests it should (we are comparing execution time in different languages after all).

However, if I modify the code to return, say, the count in the spawned thread (and later do `.join().unwrap()`, the execution time starts depending on the number of iterations.

This seems not to depend on the number of iterations:

    pub extern fn process() {
        let handles: Vec<_> = (0..10).map(|_| {
            thread::spawn(|| {
                let mut _x = 0;
                for _ in (0..900_000_001) {
                    _x += 1
                }
            })
        }).collect();
    
        for h in handles {
            h.join().unwrap();
        }
    }

This seems to get slower with more iterations (as it should):

    pub extern fn process() {
        let handles: Vec<_> = (0..10).map(|_| {
            thread::spawn(|| {
                let mut _x = 0;
                for _ in (0..900_000_001) {
                    _x += 1
                }
                _x  // This line makes execution time depend on the number of iterations
            })
        }).collect();

        for h in handles {
            h.join().unwrap();
        }
    }

Why is this so? Is this some sort of compiler optimisation that removes the loop execution altogether? Should the code in the book be updated?

  [1]: http://doc.rust-lang.org/stable/book/rust-inside-other-languages.html

### PR DESCRIPTION
**Pull request by jooert**
Use result of the computation to prevent the compiler from optimising
too much. Change `_x` to `x` and therefore remove the paragraph about
the underscore.

Fixes #25855.

r? @steveklabnik

### CODE REVIEW

File: src/doc/trpl/rust-inside-other-languages.md [hunk 1]
```diff
@@ -66,10 +66,15 @@ threads = []
     5_000_000.times do
       count += 1
     end
+
+    Thread.current[:output] = count
```
Comment (steveklabnik) [inline src/doc/trpl/rust-inside-other-languages.md]: hmmm, this changes it from the rust significantly, doesn't it?

Comment (jooert) [inline src/doc/trpl/rust-inside-other-languages.md]: Probably, my ruby is not very up to par. Would it be better to just return `count` from the threads and then collect the values using `t.value`?

Comment (steveklabnik) [inline src/doc/trpl/rust-inside-other-languages.md]: yeah, that seems better :+1:

**Comment by jooert** [comment]
Updated.

**Comment by steveklabnik** [comment]
Thanks! :)

**Pull request merged**